pkg/smart/ataidentify.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2018 The OpenEBS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package smart
+
+import (
+	"fmt"
+
+	"github.com/openebs/node-disk-manager/pkg/util"
+)
+
+// swapByteOrder swaps the order of every second byte in a byte slice and modifies the slice in place.
+func (d *ATACSPage) swapByteOrder(b []byte) []byte {
+	tmp := make([]byte, len(b))
+
+	for i := 0; i < len(b); i += 2 {
+		tmp[i], tmp[i+1] = b[i+1], b[i]
+	}
+	return tmp
+}
+
+// getSerialNumber returns the serial number of a disk device using an ATA IDENTIFY command.
+func (d *ATACSPage) getSerialNumber() []byte {
+	return d.swapByteOrder(d.SerialNumber[:])
+}
+
+// getWWN returns the worldwide unique name for a disk
+func (d *ATACSPage) getWWN() string {
+	NAA := d.WWN[0] >> 12
+	IEEEOUI := (uint32(d.WWN[0]&0x0fff) << 12) | (uint32(d.WWN[1]) >> 4)
+	UniqueID := ((uint64(d.WWN[1]) & 0xf) << 32) | (uint64(d.WWN[2]) << 16) | uint64(d.WWN[3])
+
+	return fmt.Sprintf("%x %06x %09x", NAA, IEEEOUI, UniqueID)
+}
+
+// getSectorSize returns logical and physical sector sizes of a disk
+func (d *ATACSPage) getSectorSize() (uint32, uint32) {
+	// By default, we are assuming the physical and logical sector size to be 512
+	// based on further check conditions, it would be altered.
+	var LogSec, PhySec uint32 = 512, 512
+
+	if (d.SectorSize & 0xc000) != 0x4000 {
+		return LogSec, PhySec
+	}
+	// TODO : Add support for Long Logical/Physical Sectors (LLS/LPS)
+	if (d.SectorSize & 0x2000) != 0x0000 {
+		// Physical sector size is multiple of logical sector size
+		PhySec <<= (d.SectorSize & 0x0f)
+	}
+	return LogSec, PhySec
+}
+
+// getATAMajorVersion returns the ATA major version of a disk using an ATA IDENTIFY command.
+func (d *ATACSPage) getATAMajorVersion() (s string) {
+	if (d.MajorVer == 0) || (d.MajorVer == 0xffff) {
+		return "This device does not report ATA major version"
+	}
+	majorVer := util.MSignificantBit(uint(d.MajorVer))
```

Comment:
we can avoid switch case by using if we can create prefix group like
ATA range(1-3)
```
majorVerATAPrefix := "ATA-"
majorVer := util.MSignificantBit(uint(d.MajorVer))
if majorVer <= 3{
return majorVerATAPrefix + string(majorVer)
}
```
ATA-ATAPI range (4-7)
ACS range (9-10)
ATA8-ACS (8)

Reply:
Or we can use groups inside switch case.